app/src/main/dist/bin/crate

```diff
@@ -181,20 +179,23 @@ while true; do
             echo "  -X            set a nonstandard java option"
             exit 0
         ;;
-        -D)
-            javaProperties="$javaProperties -D$2"
-            shift 2
+        D)
+            javaProperties="$javaProperties -D$OPTARG"
         ;;
-        -C)
-            properties="$properties -C$2"
-            shift 2
+        C)
+            set -f  # disable glob
+            arg="-C$OPTARG"
+            # Add quotes to optarg if contains '='
+            if [[ $arg == *"="* ]]; then
```

Comment:
Please check if that works with `dash` for example.
Maybe you can do: `arg=$(echo $OPTARG | sed 's/=/=\"/g' | sed 's/$/\"/g')` without the if